modules/couchbase/storage_mode.go

```diff
@@ -0,0 +1,9 @@
+package couchbase
+
+type indexStorageMode string
+
+const (
+	MemoryOptimized indexStorageMode = "memory_optimized"
```

Comment:
Could we copy the comments from the Java PR here, so each constant is self-explanatory with the comments? 🙏

Reply:
Done 👍